File: app/src/main/java/com/paavam/todoapp/adapter/NotesAdapter.java
package com.paavam.todoapp.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.paavam.todoapp.R;
import com.paavam.todoapp.interfaces.NotesListener;
import com.paavam.todoapp.model.Note;

import java.util.List;

public class NotesAdapter extends RecyclerView.Adapter<NotesAdapter.ViewHolder> {

    private List<Note> noteList;
    private NotesListener listener;

    public NotesAdapter(List<Note> noteList) {
        this.noteList = noteList;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.notes_item_row, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        final Note note = noteList.get(position);

        holder.title.setText(note.getTitle());
        holder.desc.setText(note.getDescription());

        holder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                listener.onClickedItem(note);
            }
        });
    }

    @Override
    public int getItemCount() {
        return noteList.size();
    }

    public void setNotesListener(NotesListener listener) {
        this.listener = listener;
    }

    static class ViewHolder extends RecyclerView.ViewHolder {
        private TextView title, desc;

        ViewHolder(@NonNull View itemView) {
            super(itemView);
            title = itemView.findViewById(R.id.title_item);
            desc = itemView.findViewById(R.id.desc_item);
        }
    }
}

File: app/src/main/java/com/paavam/todoapp/adapter/NotesAdapter.kt
package com.paavam.todoapp.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.paavam.todoapp.R
import com.paavam.todoapp.db.Note
import com.paavam.todoapp.interfaces.NotesListener

public class NotesAdapter(private val allNotesList: List<Note>, private val listener: NotesListener) : RecyclerView.Adapter<NotesAdapter.ViewHolder>() {

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val title = itemView.findViewById<TextView>(R.id.title_item)
        val desc = itemView.findViewById<TextView>(R.id.desc_item)
    }


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.notes_item_row, parent, false)
        return ViewHolder(view)
    }

    override fun getItemCount(): Int {
        return allNotesList.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val note = allNotesList[position]

        holder.title.text = note.title
        holder.desc.text = note.description

        holder.itemView.setOnClickListener { listener.onClickedItem(note) }
    }

}